THEMING.md

```diff
@@ -0,0 +1,134 @@
+<!--docs:
+title: "Theming"
+layout: detail
+section: components
+excerpt: "Material Theming refers to the customization of your Material Design app to better reflect your product’s brand."
+-->
+
+# Theming
+
+Material Theming refers to the customization of your Material Design app to better reflect your product’s brand.
+
+## Overview
+
+### Terminology
+
+Our approach to theming relies on the relationships between the following concepts:
+
+1. Components
+2. Subsystem Schemes
+3. The Container Scheme
+4. Theming Extensions
+
+Components are expected to provide public APIs for a variety of parameters. An example of a component is [MDCButton](https://github.com/material-components/material-components-ios/tree/develop/components/Buttons).
+
+Subsystem schemes represent a set of opinionated properties that are intended to be mapped to component parameters. There is a scheme for each Material Theming subsystem. For example, there is a scheme for the color, shape, and typography subsystems.
+
+The Container scheme represents a single configurable entity that is applicable to all themeable components. A container scheme consists of all of the subsystem schemes.
+
+Theming extensions are component extensions that, when invoked with a default container scheme, will theme a component according to the [Material Design guidelines](https://material.io/design). When provided with subsystem schemes via a container scheme, the extension will map the subsystem scheme's values to the component’s parameters.
+
+### Sensible defaults, yet highly configurable
+
+By default, we try and give our components sensible values for all of their customizable properties, things such as `backgroundColor` or `titleFont`. You can use these defaults as a baseline, but we encourage you to theme your components to match your brand style.
+
+### Schemes
+
+<ul class="icon-list">
+<li class="icon-list-item icon-list-item--link"><a href="Container/">Container scheme</a></li>
+<li class="icon-list-item icon-list-item--link"><a href="Color/">Color scheme</a></li>
+<li class="icon-list-item icon-list-item--link"><a href="Shape/">Shape scheme</a></li>
+<li class="icon-list-item icon-list-item--link"><a href="Typography/">Typography scheme</a></li>
+</ul>
+
+## Examples
+
+### Using a subsystem scheme
```

Comment:
When listing examples in documentation it can be helpful to list them in the order of most-commonly-expected use. I think that subsystem schemes will be fairly uncommonly used in relation to customizing container schemes, which in turn I expect to be used less than the theming extensions themselves; can we consider ordering the examples in the reverse of their current order?

Reply:
Done